core-api-library/library/src/main/java/net/gini/android/core/api/RetrofitHelper.kt

```diff
@@ -0,0 +1,52 @@
+package net.gini.android.core.api
+
+import net.gini.android.core.api.authorization.UserService
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import java.util.concurrent.TimeUnit
+
+object RetrofitHelper {
+    private var baseUrl = ""
+    private const val DEFAULT_TIMEOUT = 60L
+
+    val setAuthorizationHeaderUseCase = SetAuthorizationHeaderUseCase()
+
+    fun getInstance(): Retrofit {
+        return Retrofit.Builder()
+            .client(
+                OkHttpClient.Builder()
+                .connectTimeout(DEFAULT_TIMEOUT, TimeUnit.SECONDS)
+                .readTimeout(DEFAULT_TIMEOUT, TimeUnit.SECONDS)
+                .writeTimeout(DEFAULT_TIMEOUT, TimeUnit.SECONDS)
+                    .addInterceptor {
+                        it.proceed(setAuthorizationHeaderUseCase(it.request()))
+                    }.build())
+            .baseUrl(baseUrl)
+            .addConverterFactory(GsonConverterFactory.create())
+            .build()
+    }
+
+    fun provideUserService(): UserService {
+        return getInstance().create(UserService::class.java)
+    }
+
+    fun setBaseURL(baseURL: String) {
+        baseUrl = baseURL
+    }
+
+    class SetAuthorizationHeaderUseCase {
```

Comment:
Setting the authorisation header has to happen separately for each request. Some requests need basic auth and others bearer tokens.